docs/pages/guides/spend-permissions/api-reference.mdx

```diff
@@ -0,0 +1,276 @@
+Reverts if the batch is empty.
+
+```solidity
+function getBatchHash(SpendPermissionBatch memory spendPermissionBatch) public view returns (bytes32);
+```
+
+---
+
+#### `isApproved`
+
+Check if a spend permission is approved and not revoked.
+
+```solidity
+function isApproved(SpendPermission memory spendPermission) public view returns (bool);
+```
+
+---
+
+#### `getCurrentPeriod`
+
+Retrieve the start, end, and accumulated spend for the current period of a spend permission.
+
+```solidity
+function getCurrentPeriod(SpendPermission memory spendPermission) public view returns (PeriodSpend memory);
+```
+## Coinbase Smart Wallet clients
+
+### Using Viem
+[Viem](https://viem.sh/) is a TypeScript interface for Ethereum that provides low-level stateless primitives for interacting with Ethereum.
+
+- **[Getting Started with Viem](https://viem.sh/docs/getting-started.html)** 
+
+- **[Getting Started with Account Abstraction](https://viem.sh/account-abstraction#getting-started-with-account-abstraction)** 
+   Viem provides support for account abstraction clients, including specific support for Coinbase Smart Wallet and paymasters.
+
+### Using OnchainKit
+[OnchainKit](https://onchainkit.xyz/) is a collection of React components and TypeScript utilities that help developers quickly build
+onchain applications.
+
+### Using Wagmi
+[Wagmi](https://wagmi.sh/) is a collection of React Hooks that facilitate development of blockchain frontends.
+
+-  **[Getting Started with Wagmi](https://wagmi.sh/react/getting-started)** 
+- **[coinbaseWallet](https://wagmi.sh/react/api/connectors/coinbaseWallet)** 
+
+
+## Coinbase Wallet API
+
+### `wallet_fetchPermissions`
+
+#### Schema
+```typescript
+type FetchPermissionsRequest = {
+  account: Address;
+  chainId: string; // 0x...
+  spender: Address;
+  pageOptions: PageOptions; // optional, defaults to page size 50 and empty cursor
+}
+
+type FetchPermissionsResponse = {
+  permissions: FetchPermissionsResponseItem[];
+  pageDescription: PageDescription;
+}
+
+type PageOptions = {
+  pageSize number; // number of items requested, defaults to 50
+  cursor string; // identifier for where the page should start
+}
+
+type PageDescription = {
+  pageSize number; // number of items returned
+  nextCursor string; // identifier for where the next page should start
+}
+
+type FetchPermissionsResponseItem = {
+  createdAt: number; // UTC timestamp for when the permission was granted
+  permissionHash: string;
+  signature: string;
+  permission: {
+    account: string;
+    spender: string;
+    token: string;
+    allowance: string; // base 10 numeric string
+    period: number;
+    start: number;
+    end: number;
+    salt: string; // base 10 numeric string
+    extraData: string // 0x...
+  };
+}
+```
+
+#### Example request
+
+```bash
+curl --location 'https://wallet-chain-proxy-dev.cbhq.net?targetName=base-sepolia' \
```

Comment:
Gut says we shouldn't be exposing this endpoint, can you confirm with Sam?

Reply:
This is our internal dev endpoint, for now let's keep this out of the docs since devs won't be able to hit this externally until we launch to prod. For reference, this will eventually be `https://chain-proxy.wallet.coinbase.com/?targetName=base-sepolia` though.

Reply:
great, just pulling out the entire example request/response for now